plugins/optimization-detective/prime-url-metrics-block-editor.js

```diff
@@ -0,0 +1,178 @@
+	iframe.id = 'od-prime-url-metrics-iframe';
+	iframe.style.position = 'fixed';
+	iframe.style.top = '0';
+	iframe.style.left = '0';
+	iframe.style.transform = 'scale(0.05)';
+	iframe.style.transformOrigin = '0 0';
+	iframe.style.pointerEvents = 'none';
+	iframe.style.opacity = '0.000001';
+	iframe.style.zIndex = '-9999';
+	document.body.appendChild( iframe );
+
+	/**
+	 * Primes the URL metrics for all breakpoints.
+	 * @return {Promise<void>} The promise that resolves to void.
+	 */
+	async function processTasks() {
+		try {
+			isProcessing = true;
+			if ( 0 === breakpoints.length ) {
+				breakpoints = await apiFetch( {
+					path: '/optimization-detective/v1/prime-urls-breakpoints',
+					method: 'GET',
+				} );
+			}
+
+			const permalink = select( 'core/editor' ).getPermalink();
+			verificationToken = await apiFetch( {
+				path: '/optimization-detective/v1/prime-urls-verification-token',
+				method: 'GET',
+			} );
+
+			currentTasks = breakpoints.map( ( breakpoint ) => ( {
+				url: permalink,
+				width: breakpoint.width,
+				height: breakpoint.height,
+			} ) );
+
+			while ( isProcessing && currentTaskIndex < currentTasks.length ) {
+				await processTask(
+					currentTasks[ currentTaskIndex ],
+					abortController.signal
+				);
+				currentTaskIndex++;
+			}
+			isProcessing = false;
+		} catch ( error ) {
+			isProcessing = false;
+		}
+	}
+
+	/**
+	 * Loads the iframe and waits for the message.
+	 * @param {{url: string, width: number, height: number}} task   - The breakpoint to set for the iframe.
+	 * @param {AbortSignal}                                  signal - The signal to abort the task.
+	 * @return {Promise<void>} The promise that resolves to void.
+	 */
+	async function processTask( task, signal ) {
+		return new Promise( ( resolve, reject ) => {
+			const handleMessage = ( event ) => {
+				if ( 'OD_PRIME_URL_METRICS_REQUEST_SUCCESS' === event.data ) {
+					cleanup();
+					resolve();
+				}
+			};
+
+			const abortHandler = () => {
+				cleanup();
+				reject( new Error( 'Task Aborted' ) );
+			};
+
+			const cleanup = () => {
+				signal.removeEventListener( 'abort', abortHandler );
+				window.removeEventListener( 'message', handleMessage );
+				clearTimeout( timeoutId );
+				iframe.onerror = null;
+				iframe.src = 'about:blank';
+			};
+
+			const timeoutId = setTimeout( () => {
+				cleanup();
+				reject( new Error( 'Timeout waiting for message' ) );
+			}, 30000 ); // 30-second timeout
+
+			if ( signal.aborted ) {
+				abortHandler();
+				return;
+			}
+
+			signal.addEventListener( 'abort', abortHandler );
+			window.addEventListener( 'message', handleMessage );
+
+			iframe.onerror = () => {
+				cleanup();
+				reject( new Error( 'Iframe failed to load' ) );
+			};
+
+			// Load the iframe
+			iframe.src = task.url;
+			iframe.width = task.width.toString();
+			iframe.height = task.height.toString();
+			iframe.dataset.odPrimeUrlMetricsVerificationToken =
+				verificationToken;
```

Comment:
See https://github.com/WordPress/performance/pull/1850/files#r2038090196

I suggest adding the token as a query parameter for the URL being loaded in the iframe.